Update price limits in OfferMapper

ahmedNaser7 opened the pull request:
Update price limits in OfferMapper

The `price` property in `OfferDto.toOffer()` and `Offer.toOfferDto()` is now constrained to be within a minimum of 0.0 and a maximum of 900000.0 using the `coerceIn` function.

<img width="720" height="1200" alt="Screenshot_20250817_120719" src="https://github.com/user-attachments/assets/54862460-3532-4fe3-9dc6-8d30d3ed7ab9" />

ahmedNaser7 committed:
Update price limits in OfferMapper

The `price` property in `OfferDto.toOffer()` and `Offer.toOfferDto()` is now constrained to be within a minimum of 0.0 and a maximum of 900000.0 using the `coerceIn` function.

Joseph-Sameh-0 commented on app/src/main/java/com/paris_2/san3a/data/mapper/OfferMapper.kt:
```diff
@@ -7,7 +7,7 @@ fun Offer.toDto()  = OfferDto(
     id = id,
     requestId = requestId,
     craftsmanId = craftsmanId,
-    price = price,
+    price = price.coerceIn(MinPrice,MaxPrice),
```
not in the mapper 
prevent the user from adding this value in add offer section

ahmedNaser7 replied:
nah , i only let him enter his offer as he like and only the max value i set will accept

Joseph-Sameh-0 replied:
it's not good
if you let me entered a big number don't change it
instead of this you can prevent the user from enter this number

ahmedNaser7 replied:
i got you , i will handle it in add offer state and set the limit of digit in view model

ahmedNaser7 committed:
fix:limit the length of the price input to a maximum of 7 digits
 and remove coercion in offer price mapping and limit price length

This commit removes the price coercion between `MinPrice` and `MaxPrice` in `OfferMapper.kt` when converting `OfferUiState` to `Offer` and `OfferDto` to `Offer`.

Additionally, the `CraftsmanRequestDetailsViewModel` now limits the length of the price input to a maximum of 7 digits.

ahmedNaser7 committed:
Refactor: extract price validation logic to a separate function

This commit refactors the `onPriceChanged` function in `CraftsmanRequestDetailsViewModel.kt` by extracting the input price validation logic into a new private function `isInputPriceValid`. This improves code readability and maintainability.

ahmedNaser7 committed:
Merge branch 'develop' into bug-fix/fix-money-limit

ahmedNaser7 committed:
Merge branch 'develop' into bug-fix/fix-money-limit
